rpc/src/client/subscription.rs

```diff
@@ -241,9 +242,11 @@ pub struct SubscriptionRouter {
     subscriptions: HashMap<String, HashMap<SubscriptionId, ChannelTx<Result<Event>>>>,
     // A map of JSON-RPC request IDs (for `/subscribe` requests) to pending
     // subscription requests.
+    #[cfg(feature = "websocket-client")]
```

Comment:
Can't we just put these on the struct instead of the fields?

Reply:
So the need for this comes from all those annoying clippy warnings we keep getting, and they only come up when the `default` or `http-client` features are active (but they go away when the `websocket-client` feature is present).

I'll experiment with having two separate `impl` blocks for the same struct and adding the feature attribute to the WebSocket client-specific `impl` block. Haven't tried that yet.

Reply:
I've refactored this a bit, separating out standard `SubscriptionRouter` functionality from `TwoPhaseSubscriptionRouter` functionality.

The two-phase subscription router is a composition of a `SubscriptionRouter` that enhances it with two-phase addition/removal of subscriptions. This makes it simpler to delineate which parts are only relevant for the `websocket-client` feature.